### ISSUE TITLE
Incomplete SAML Attribute Mapping in Teleport for Multiple Assumed Roles

### ISSUE THREAD

**Issue by programmerq** [gravitational/teleport#40734]
### Expected behavior:
When multiple roles are assumed in Teleport, all the roles should be sent as attributes in the roles SAML assertion.

### Current behavior:
Only one of the multiple assumed roles appears in the SAML assertion. When two roles are assumed, the SAML assertion to the SP only includes one role and omits the other. All of the user's normal roles are still present. It only allows for one role to appear instead of all of the assumed roles.

### Bug details:
- **Teleport version**: 15.x
- **Recreation steps**:
  1. User assumes one or more access requests in the Teleport Web UI, so they have at least two roles assumed.
  2. User expects all assumed roles to be passed as attributes in a SAML assertion.
  3. Only one assumed role is included in the SAML assertion.

- **Debug logs**:
  - Given this screenshot (manually redacted to generic values) that shows multiple assumed roles:
![screenshot redacted](https://github.com/gravitational/teleport/assets/37278/b2989bad-ffa0-4a2c-89ef-c0391f87952c)
  - The following SAML assertion looks like this snippet:
  ```xml
  <saml:AttributeStatement>
  <saml:Attribute FriendlyName="role" Name="urn:example:attributes:role" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname—format:uri">
    <saml:AttributeValue ...>developers</saml:AttributeValue>
    <saml:AttributeValue ...>contractors</saml:AttributeValue>
    <saml:AttributeValue ...>db-admin</saml:AttributeValue>
    <!-- Missing AttributeValue for qa-admin -->
  </saml:Attribute>
  </saml:AttributeStatement>
  ```

edit1: The first version indicated the issue was that it would include roles from only one access request. The actual issue is that it only includes exactly one role, even if a single access request has multiple roles.

**Comment by programmerq** [comment]
I believe that this bug is a quirk due to how access requests are implemented. My understanding is that you can only really have one assumed access request at a time. This is clear when assuming roles in a `tsh` session. I think the Web UI gets around this by keeping track of which requests you have asked to assume, and then grabs a single matching request when you access an SSH service or view a session recording, etc? That works for most of the resources accessed via the Web UI, but with the SAML idp, it makes sense that you would expect all the roles listed in the Web UI to be included.

**Comment by flyinghermit** [comment]
Chatted with @programmerq in Slack but writing here for visibility. 

I was not able to reproduce this issue. 
I tested with two assumed roles and both of the assumed roles were included in SAML assertion. 
Tested on `v15.4.0` and latest alpha release `v16.0.0-alpha.3`

A rough video I recorded that shows two assumed roles passed in SAML assertion:

https://github.com/gravitational/teleport/assets/10177602/35a21080-616e-4def-bca9-ff4127c96037
